app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,15 @@
+class CommentsController < ApplicationController
+  def create
+    user_id = params.require(:post).permit(:user_id)
+    user = User.where(id: user_id['user_id']).first
+    post = Post.find(params[:post_id])
+    @comment = Comment.new(params.require(:post).permit(:text, :post_id, :user_id))
+
+    if @comment.save
+      redirect_to "/users/#{user.id}/posts/#{post.id}", notice: 'Saved successfully'
+    else
+      flash[:notice] = 'Whoops!! Please try again. Something went wrong'
+      render 'posts/index'
+    end
+  end
+end
```

Comment:
- [ ] Kindly add the `new` method to initialize a comment

Reply:
Hi, thank you for your review. Could kindly explain some more.
Because I have this line `@comment = Comment.new(params.require(:post).permit(:text, :post_id, :user_id))` in the comments controller which creates an instance of the Comments model class. 

Also, my approach to create a comment is that I have the comments form already displayed when the user clicks on the post (visits post details), right beneath the post comments. I do not have a button that a user needs to click to display the form. Given this approach, do i still need to add the method `new`?

Reply:
> * [ ]  Kindly add the `new` method to initialize a comment

Hi, thank you for your review. Could kindly explain some more.
Because I have this line `@comment = Comment.new(params.require(:post).permit(:text, :post_id, :user_id))` in the comments controller which creates an instance of the Comments model class.

Also, my approach to create a comment is that I have the comments form already displayed when the user clicks on the post (visits post details), right beneath the post comments. I do not have a button that a user needs to click to display the form. Given this approach, do i still need to add the method `new`?